src/platform_impl/linux/x11/util/input.rs

```diff
@@ -25,16 +25,16 @@ impl From<ffi::XIModifierState> for ModifiersState {
 
 pub struct PointerState<'a> {
```

Comment:
It turns out that some of these private fields are unused. rustc wasn't detecting this before because the struct was being filled with `&mut pointer_state.field` references and this was seen as a use. I've commented out the fields as it should have no effect on the program. I could instead make them public or add `#[allow(dead_code)]` to the struct.

Reply:
I kept them in both 1) to make allocation easier, and 2) so that contributors can more easily discover/utilize functionality they might need (the idea is to provide a robust internal API). I'd prefer they be kept in, though with comments explaining the rationale.

Reply:
Fair enough. I've restored the fields and marked them `pub` to silence the warnings.